pkgs/development/tools/kafkacat/default.nix

```diff
@@ -0,0 +1,25 @@
+{ stdenv, fetchFromGitHub, zlib, perl, pkgconfig, rdkafka, yajl }:
+
+stdenv.mkDerivation rec {
+  name = "kafkacat-${version}";
+  version = "1.3.1";
+
+  src = fetchFromGitHub {
+    owner = "edenhill";
+    repo = "kafkacat";
+    rev = "${version}";
+    sha256 = "1fgs04rclgfwri6vd9lj0mw545nmscav9p6kh7r28k5ap2g0gak5";
+  };
+
+  buildInputs = [ zlib perl pkgconfig rdkafka yajl ];
```

Comment:
perl is not needed. pkgconfig should go to nativeBuildInputs.

Reply:
Why `nativeBuildInputs`? I've never clearly understood the distinction.

Reply:
https://nixos.org/nixpkgs/manual/#ssec-stdenv-attributes see `nativeBuildInputs`.

Reply:
It is important for cross-compiling, where you dependencies like a compiler build for the current architectures and libraries which are compiled for the target architecture.

Reply:
How would I have figured out that pkgconfig should go in nativeBuildInputs?

Candidly, I started by copying the rdkafa expression, which is why perl is in there and pkgconfig is a regulart buildInput

Reply:
All programs that run during the build are `nativeBuildInputs`, and basically all libraries are `buildInputs`. `pkgconfig` is a program.

Until recently `pkgconfig` was explicitly automatically promoted from `buildInputs` into `nativeBuildInputs` so there was no practical difference. Since #29584 it is no longer the case, and packages with `pkgconfig` in `buildInputs` are broken for cross compilation.